src/Hosting/Hosting/test/StaticWebAssets/StaticWebAssetsFileProviderTests.cs

```diff
@@ -37,6 +38,49 @@ public void StaticWebAssetsFileProvider_Constructor_DoesNotPrependPrefixWithSlas
             Assert.Equal("/_content", provider.BasePath);
         }
 
+        [Theory]
+        [InlineData("", "_content")]
+        [InlineData("/_content", "RazorClassLib")]
+        [InlineData("/_content/RazorClassLib", "Dir")]
+        public void GetDirectoryContents_WalksUpContentRoot(string searchDir, string expected)
```

Comment:
Can you add a couple of cases that validate that it hands off correctly the rest of the path to the physical file system?